ch02.adoc

```diff
@@ -268,3 +268,165 @@ They may not be attached to a group, even if all variables within that group use
 If attributes are present within groups without being attached to a variable, these attributes apply to the group where they are defined, and to that group's descendants, but not to ancestor or sibling groups.
 If a group attribute is defined in a parent group, and one of the child group redefines the same attribute, the definition within the child group applies for the child and all of its descendants.
 
+[[aggregation-variables, Section 2.8, "Aggregation Variables"]]
+=== Aggregation Variables
+
+An __aggregation variable__ is a variable which has been formed by combining (i.e. aggregating) multiple __fragments__ stored in __fragment files__ that are external to the file containing the aggregation variable, i.e. the __aggregation file__. The aggregation variable does not contain any actual data, instead it contains instructions on how to create its __aggregated data__ as an aggregation of the data from each fragment.
+
+Aggregation provides the utility of being able to view, as a single entity, a dataset that has been partitioned across multiple files, whilst taking up very little space on disk (since the aggregation file contains no copies of the data in the fragments). The fragment files may be CF-compliant or have any other format, thereby allowing an aggregation variable to act as CF-compliant view of non-CF datasets. Storing aggregations is useful for data analysis, as it avoids the computational expense of deriving the aggregation at the time of analysis; and for archive curation, as the aggregation can act as a metadata-rich archive index.
+
+An aggregation variable must be a scalar (i.e. it has no dimensions) and the value of its single element is immaterial. It acts as a container for all of the usual attributes that describe the data, with the addition of two special attributes: one that defines the _aggregated dimensions_, i.e. the dimensions of the aggregated data; and one that provides the instructions on how the aggregated data is to be created. The data type of the aggregated data is the same as the data type of the aggregation variable.
+
+Any variable may be an aggregation variable, and being an aggregation variable does not affect its role within CF (i.e. data variable, coordinate variable, boundary variable, cell measure variable, etc.).
+
+The details of how to encode and decode aggregation variables are given in this section, with examples provided in <<appendix-aggregation-examples>>.
+
+The conventions do not currently offer guidance to dataset creators on how to decide if two or more fragments can be aggregated in this way.
```

Comment:
But there are conventions, you're simply saying that they are not internal to _the_ CF conventions?

Reply:
Yes - "The conventions" means "The CF conventions" (should be explicit about that), and there are multiple external rule-sets for deciding if two or more fragments can be aggregated (xarray, iris, cf-python, Ezequiel's, etc.), but none of them are officially part of CF. All of these rule sets are more or less implementing the same set intuitive rules (e.g. "only things with the same identity and equivalent units can be aggregated"). The [cf-python rules](https://ncas-cms.github.io/cf-python/aggregation_rules.html) that Jonathan and I devised are certainly the most complete, as they are based on the complete CF data model.